src/dependency.rb

```diff
@@ -122,6 +122,9 @@ def valid_file?(dep, value, validate_directory = false, &block)
 
       if validate_directory
         Dir.exist?(path)
+      # The orphan group's answer file is always optional
+      elsif (path == FilePath.group_answers('orphan')) && !File.file?(path)
```

Comment:
This line is very dense - I'd suggest breaking it down into descriptively named variables or methods; if you can name these well enough you might be able to remove the comment as well as it will be obvious what's going on just from the code.

More generally, CodeClimate is now complaining about the complexity of this method, and I agree with it. The natural place to split it seems like it would be to at least split out methods for the `case` statement above and this `if` statement, but feel free to split it up however you want.

Reply:
Yeah, I noticed that when I created the PR however I wouldn't put to much effort in fixing this issue per se.

The `Dependency` class as a whole is quite old code and I don't think a lot of it is relevant anymore. We are switching to a model of validation on the fly, so it is starting to become redundant.

Now that we support building non-groups, we can't validate the group answer file exists. It maybe worth going over `Dependency` and stripping back some of the functionality.

Alternatively, we could move the answer specific validation into the `Loader` class (in addition to the validation it already does).

Reply:
:+1: